libraries/exoplayer_rtsp/src/main/java/androidx/media3/exoplayer/rtsp/reader/RtpVp9Reader.java

```diff
@@ -85,19 +91,19 @@ public void createTracks(ExtractorOutput extractorOutput, int trackId) {
   }
 
   @Override
-  public void onReceivingFirstPacket(long timestamp, int sequenceNumber) {}
+  public void onReceivingFirstPacket(long timestamp, int sequenceNumber) {
+    checkState(firstReceivedTimestamp == C.TIME_UNSET);
+    firstReceivedTimestamp = timestamp;
+  }
 
   @Override
   public void consume(
       ParsableByteArray data, long timestamp, int sequenceNumber, boolean rtpMarker) {
     checkStateNotNull(trackOutput);
 
     if (validateVp9Descriptor(data, sequenceNumber)) {
-      @C.BufferFlags int bufferFlags = 0;
-      if (fragmentedSampleSizeBytes == 0
-          && gotFirstPacketOfVP9Frame
-          && (data.peekUnsignedByte() & 0x04) == 0) {
-        bufferFlags = C.BUFFER_FLAG_KEY_FRAME;
+      if (fragmentedSampleSizeBytes == 0 && gotFirstPacketOfVP9Frame) {
+        isKeyFrame = (data.peekUnsignedByte() & 0x04) == 0;
```

Comment:
I'm assuming you are parsing the VP9 payload descriptor's E bit, which is 

> E:  End of a frame.  MUST be set to 1 for the final RTP packet of a
      VP9 frame, and 0 otherwise.  This enables a decoder to finish
      decoding the frame, where it otherwise may need to wait for the
      next packet to explicitly know that the frame is complete.  Note
      that, if spatial scalability is in use, more frames from the same
      picture may follow; see the description of the B bit above.

I think this bit equals to zero is not the sufficient condition that the current frame is a key frame, no?

Reply:
SG, thanks